chain/types/ethtypes/eth_types.go

```diff
@@ -45,6 +45,13 @@ func (e *EthUint64) UnmarshalJSON(b []byte) error {
 	if err := json.Unmarshal(b, &s); err != nil {
 		return err
 	}
+	if !startsWith0x(s) {
+		return fmt.Errorf("EthNonce UnmarshalJSON on %s failed, must start with 0x", s)
+	}
+	if hasLeadingZeroes(s) {
+		return fmt.Errorf("EthUint64 UnmarshalJSON on %s failed, must not have a leading 0x0 before the value", s)
+	}
```

Comment:
I don't think it's worth enforcing, inevitably some random important library will do this wrong.

Reply:
it feels icky at this point to merge -- but i'm following through with the ask from @jennijuju and making the PR

Reply:
Agree with the assessment here. I'd rather us be more lenient on the _input_ side of things. I am checking with other clients to confirm if they apply such strictness here though.

Reply:
So other popular clients are truly enforcing the 0x prefix for incoming Bigints and Uint64, and only allow the first character to be a zero when string length = 1 (i.e. no zero prefixes, indeed).